### PR TITLE
Introduced IRequest interface for CFamily subprocess requests

### PR DESCRIPTION
**Pull request by duncanp-sonar**
* changed CLangAnalyzer to use the new interface
* fixed up tests

Prep for #2194

### CODE REVIEW

File: src/Integration.Vsix.UnitTests/CFamily/CFamilyHelperTests.cs [hunk 1]
```diff
@@ -120,6 +120,7 @@ public void CreateRequest_NoAnalyzerOptions_RequestCreatedWithoutOptions()
             (request.Flags & Request.BuildPreamble).Should().Be(0);
             request.RulesConfiguration.Should().NotBeNull();
             request.Options.Should().NotBeEmpty();
+            request.AnalyzerOptions.Should().BeNull();
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/CFamily/CFamilyHelperTests.cs:123]: This file contains for "create request". I've fixed them up to check that `IRequest.AnalyzerOptions` is set appropriately.

File: src/Integration.Vsix.UnitTests/CFamily/CFamily_CallAnalyzerTests.cs [hunk 2]
```diff
@@ -40,7 +42,7 @@ public void CallAnalyzer_Succeeds_ReturnsMessages()
             var dummyProcessRunner = new DummyProcessRunner(MockResponse());
 
             // Act
-            var response = GetResponse(dummyProcessRunner, new Request(), new TestLogger());
+            var response = GetResponse(dummyProcessRunner, CreateRequest(), new TestLogger());
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/CFamily/CFamily_CallAnalyzerTests.cs:45]: The "call analyzer" tests no longer use the `Request` class - they use the interface instead.

File: src/Integration.Vsix/CFamily/CFamilyHelper.cs [hunk 3]
```diff
@@ -176,17 +178,19 @@ private static void SaveFileConfig(FileConfig fileConfig, ILogger logger)
                 {
                     using (var binaryWriter = new BinaryWriter(writer.BaseStream))
                     {
-                        Protocol.Write(binaryWriter, request);
+                        // TODO: need to handle different types of IRequest
+                        var vcxProjRequest = (Request)request;
+                        Protocol.Write(binaryWriter, vcxProjRequest);
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/CFamily/CFamilyHelper.cs:183]: I've changed the methods related to calling the subprocess and handling the response to use the new interface.
This is the only place where we need to downcast.

**Comment by duncanp-sonar** [comment]
There are two new TODOs;
1) changing the creation of the request to handle different types of project
2) writing the appropriate protocol for the request when triggering the analysis.

File: src/Integration.Vsix/CFamily/Request.cs [hunk 4]
```diff
@@ -23,15 +23,14 @@
 namespace SonarLint.VisualStudio.Integration.Vsix.CFamily
 {
     // Most of the Request class is ported from Java - see PortedFromJava\Request.cs
-    // This partial contains additional properties that don't appear in the Java version
-    // and aren't passed as part of the request to the CLang analyzer, but are used by SLVS
-    // when filtering the returned issues.
-    internal partial class Request
+    internal partial class Request : IRequest
     {
         // Note: the language and RulesConfiguration aren't passed as part of the request to the
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/CFamily/Request.cs:28]: Minor: this comment can be removed as it's in the interface

**Review by rita-sonarsource — APPROVED** [review]

**Pull request merged**